subsys/net/ip/icmp.c

```diff
@@ -0,0 +1,596 @@
+/** @file
+ * @brief ICMP related functions
+ */
+
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/* Use highest log level if both IPv4 and IPv6 are defined */
+#if defined(CONFIG_NET_IPV4)
+#define MY_LOG_LEVEL CONFIG_NET_ICMPV4_LOG_LEVEL
+#endif
+
+#if defined(CONFIG_NET_IPV6)
+#if defined(MY_LOG_LEVEL)
+#if MY_LOG_LEVEL < CONFIG_NET_ICMPV6_LOG_LEVEL
+#undef MY_LOG_LEVEL
+#define MY_LOG_LEVEL CONFIG_NET_ICMPV6_LOG_LEVEL
+#endif
+#else
+#define MY_LOG_LEVEL CONFIG_NET_ICMPV6_LOG_LEVEL
+#endif
+#endif
+
+#if !defined(MY_LOG_LEVEL)
+#define MY_LOG_LEVEL LOG_LEVEL_INF
+#endif
```

Comment:
This looks a bit messy, I wonder if you can use `COND_CODE_1`:

```c
#if defined(CONFIG_NET_IPV4) && defined(CONFIG_NET_IPV6)
#define ICMP_LOG_LEVEL COND_CODE_1(CONFIG_NET_ICMPV4_LOG_LEVEL > CONFIG_NET_ICMPV6_LOG_LEVEL, \
	(CONFIG_NET_ICMPV4_LOG_LEVEL), (CONFIG_NET_ICMPV6_LOG_LEVEL))

#elif defined(CONFIG_NET_IPV4)
#define ICMP_LOG_LEVEL CONFIG_NET_ICMPV4_LOG_LEVEL
#elif defined(CONFIG_NET_IPV6)
#define ICMP_LOG_LEVEL CONFIG_NET_ICMPV6_LOG_LEVEL
#else
#define ICMP_LOG_LEVEL LOG_LEVEL_INF
#endif
```

Reply:
Cool, I will try that.

Reply:
It looks like it doesn't though :(